src/transform/src/canonicalization/flat_map_elimination.rs

```diff
@@ -42,70 +45,49 @@ impl crate::Transform for FlatMapElimination {
 }
 
 impl FlatMapElimination {
-    /// Turns `FlatMap` into `Map` if only one row is produced by flatmap.
+    /// See comment at the top of the file.
     pub fn action(relation: &mut MirRelationExpr) {
         if let MirRelationExpr::FlatMap { func, exprs, input } = relation {
-            let (func, with_ordinality) = if let TableFunc::WithOrdinality(with_ordinality) = func {
-                // get to the actual function, but remember that we have a WITH ORDINALITY clause.
-                (&*with_ordinality.inner, true)
-            } else {
-                (&*func, false)
-            };
-
-            if let TableFunc::GuardSubquerySize { .. } = func {
-                // (`with_ordinality` doesn't matter because this function never emits rows)
-                if let Some(1) = exprs[0].as_literal_int64() {
-                    relation.take_safely(None);
-                }
-            } else if let TableFunc::Wrap { width, .. } = func {
+            // Treat Wrap specially.
```

Comment:
I think this may be a good moment to record what is about to happen and why. The comment "treat wrap specially" is .. not the question a reader has at this moment, I claim, but rather "why are we treating `Wrap` specially?". Is it important that we do this and return, to ensure we do not call the rest of the function? Is it important that we don't look for `WithOrdinality` here, because it is not `Wrap` which we treat specially? Many questions that the comment doesn't address, but imo we should try and provide the context to answer in the future.